src/notation/internal/engravingitemiconengine.h

```diff
@@ -0,0 +1,45 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-Studio-CLA-applies
+ *
+ * MuseScore Studio
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2024 MuseScore Limited
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "iengravingitemiconengine.h"
+#include "engraving/rendering/isinglerenderer.h"
+
+namespace mu::notation {
+class EngravingItemIconEngine : public IEngravingItemIconEngine
```

Comment:
Since it no longer depends on QIconEngine, a better name would be EngravingItemPreviewPainter (or EngravingItemPreviewPainting, similar to NotationPainting). The word "engine" is too abstract in this context (and usually indicates that the class is doing more than it should)

Reply:
Also, I realised that it might make sense to investigate if this new class can perhaps be merged with `mu::engraving::rendering::ISingleRenderer`.